self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryInvoices.java

```diff
@@ -0,0 +1,277 @@
+package com.selfxdsd.core.mock;
+
+import com.selfxdsd.api.*;
+import com.selfxdsd.api.storage.Storage;
+import com.selfxdsd.core.contracts.invoices.StoredInvoice;
+
+import java.time.Duration;
+import java.util.*;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * In-Memory Invoices for testing purposes.
+ *
+ * @author criske
+ * @version $Id$
+ * @since 0.0.3
+ * @todo #165:30min Refactor and test ContractInvoices to be part
+ *  of self-impl API .
+ *  Also InvoiceTask should have its own concrete implementation in self-impl.
+ *  Use that in InMemoryInvoices.
+ *  Continue testing InMemoryInvoices.
+ */
+public final class InMemoryInvoices implements Invoices {
+
+    /**
+     * All contract invoices.
+     */
+    private final Map<Contract.Id, List<Invoice>> invoices = new HashMap<>();
+
+    /**
+     * Invoice Tasks. Key is the invoice id;
+     */
+    private final Map<Integer, List<InvoiceTask>> invoiceTasks =
+        new HashMap<>();
+
+    /**
+     * Active contract invoice id.
+     */
+    private final Map<Contract.Id, Integer> activeIds = new HashMap<>();
+
+    /**
+     * Payed contract invoices ids.
+     */
+    private final Map<Contract.Id, List<Integer>> payedIds = new HashMap<>();
```

Comment:
@criske I'm not sure we will need all these 4 maps. I'd like to model it so that we have only one map representing teh table, like other InMeory* implementations.

But let's leave it as it is for now, there will probably be changes anyway, later on.